scipy/optimize/_linprog_ip.py

```diff
@@ -584,8 +584,8 @@ def where(A):
             status = 0
             message = ("The solution was determined in presolve as there are "
                        "no non-trivial constraints.")
-        elif (np.any(np.logical_and(c < 0, ub == np.inf)) or
-                np.any(np.logical_and(c > 0, lb == -np.inf))):
+        elif (np.any(np.logical_and(c < 0, ub_mod == np.inf)) or
+                np.any(np.logical_and(c > 0, lb_mod == -np.inf))):
```

Comment:
Just trying to understand this line. In the example you used A_ub.size != 0, so why do you need to change here to fix the problem?  Because it seens to me that, that for passing ``test_bug_8973`` you would not need this line.

Or you just found out that you need to change here to fix other problems not included in the test case?

Reply:
Yup found that I needed this other fix. I think that I exposed this problem while debugging the original issue.

Reply:
Maybe it is worth including a test for this case...